src/v/kafka/client/direct_consumer/data_queue_cv.h

```diff
@@ -0,0 +1,37 @@
+
+
+#include <seastar/core/condition-variable.hh>
+
+namespace kafka::client::detail {
+
+// quick chop of condition variable to be better used as a fair waiter queue
+class data_queue_cv : private seastar::condition_variable {
```

Comment:
>Adds data_queue_cv which is a mask on seastar::condition_variable which
hides conditional waits

instead of masking out interfaces that aren't desired, could the user of the data_queue_cv make its CV a private member and only call the methods it needs?